### PR TITLE
102 subscription detail custom field render rtk

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#102 

- introduces a config override mechanism using RTK
- the value gets overridden when it matches a given key. Overrides take place at the app side of the project.

### CODE REVIEW

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock/WfoProductBlockKeyValueRow.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { EuiBadge } from '@elastic/eui';
+
+import { useSubscriptionDetailValueOverride } from '@/components';
+import { useWithOrchestratorTheme } from '@/hooks';
+import { FieldValue } from '@/types';
+import { camelToHuman } from '@/utils';
+
+import { getStyles } from './styles';
+
+export type WfoProductBlockKeyValueRowProps = {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock/WfoProductBlockKeyValueRow.tsx:12]: No export needed here right?

File: packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Slice, createSlice } from '@reduxjs/toolkit';
+
+import { ValueOverrideFunction } from '@/components/WfoSubscription/overrides/useSubscriptionDetailValueOverride';
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts:3]: I would move this type here instead and import it /useSubscriptionDetailValueOverride

File: packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export type ValueOverrideConfiguration = Record<string, ValueOverrideFunction>;
+
+export type OrchestratorComponentOverride = {
+    subscriptionDetail?: {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts:8]: This is fine for now but I wonder if we get more overrides if you can't make this a bit more generic. So perphaps make this something like 
```{
  [string]: valueOverrides 
}```

Keep something like that in mind when adding more overrides

**Review by DutchBen — APPROVED** [review]

**Review by Georgi2704 — APPROVED** [review]

File: packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts (hunk 3 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/rtk/slices/orchestratorComponentOverride.ts:8]: Will take this in consideration for the future overrides (next stories)

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock/WfoProductBlockKeyValueRow.tsx (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock/WfoProductBlockKeyValueRow.tsx:12]: Exporting the props type to provide override possibilities by clients

**Pull request merged**